ckanext/adfs/plugin.py

```diff
@@ -17,14 +18,9 @@
 X509 = ''
 WSFED_ENDPOINT = ''
 WTREALM = pylons.config['adfs_wtrealm']
-with open(pylons.config['adfs_federation_metadata_path']) as federation_meta:
-    METADATA = ET.fromstring(federation_meta.read())
-    for child in list(METADATA):
-        service_type = child.attrib.get('{http://www.w3.org/2001/XMLSchema-instance}type', None)
-        if service_type == 'fed:ApplicationServiceType':
-            X509 = child.find('{urn:oasis:names:tc:SAML:2.0:metadata}KeyDescriptor').xpath('string()')
-            WSFED_ENDPOINT = child.find('{http://docs.oasis-open.org/wsfed/federation/200706}PassiveRequestorEndpoint').xpath('string()')
-            break
+METADATA = get_federation_metadata(pylons.config['adfs_federation_metadata_url'])
```

Comment:
If you do this at module scope logins fail when the certificates in the federation metadata xml changes.

(Presumably until the apache process is reloaded depending on our wsgi config)